build-libs/redirects.js

```diff
@@ -155,85 +151,21 @@ async function buildDotIoRedirects() {
 		{ source: '/:path*.html', destination: '/:path*', permanent: true },
 	]
 
-	const rawVaultRedirects = isContentDeployPreview('vault')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'vault',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const vaultAuthorRedirects = eval(rawVaultRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Waypoint repo
-	// TODO: intent is to do this after all products have been migrated
-	const vaultIoRedirects = [...vaultAuthorRedirects]
-
-	const rawVagrantRedirects = isContentDeployPreview('vagrant')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'vagrant',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const vagrantAuthorRedirects = eval(rawVagrantRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Vagrant repo
-	// TODO: intent is to do this after all products have been migrated
-	const vagrantIoRedirects = [...vagrantAuthorRedirects]
-
-	const rawPackerRedirects = isContentDeployPreview('packer')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'packer',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const packerAuthorRedirects = eval(rawPackerRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Packer repo
-	// TODO: intent is to do this after all products have been migrated
-	const packerIoRedirects = [...packerAuthorRedirects]
-
-	const rawConsulRedirects = isContentDeployPreview('consul')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'consul',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const consulAuthorRedirects = eval(rawConsulRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Packer repo
-	// TODO: intent is to do this after all products have been migrated
-	const consulIoRedirects = [...consulAuthorRedirects]
-	// TODO ... consolidate redirects for other products
 	return [
 		...devPortalToDotIoRedirects,
-		...addHostCondition(boundaryIoRedirects, 'boundary'),
-		...addHostCondition(nomadIoRedirects, 'nomad'),
+		...(await getRedirectsForProduct('boundary')),
+		...(await getRedirectsForProduct('nomad')),
+		...(await getRedirectsForProduct('vault')),
```

Comment:
🤔 
```suggestion
		...(await getRedirectsForProduct('vault', 'main')),
```

Reply:
Hmm, we should probably try and derive the latest ref from the content API 💭

Reply:
...we should just make the redirects available via the content API

Reply:
> ...we should just make the redirects available via the content API

Fast-follow, post-GA

Reply:
> Hmm, we should probably try and derive the latest ref from the content API 💭

```bash
curl -s https://content.hashicorp.com/api/content/vault/version-metadata\?partial\=true \
   | jq -r '.result[0].ref'
```